app/cdap/components/WrangleHome/Components/WrangleCard/index.tsx

```diff
@@ -15,146 +15,73 @@
+    connectorTypes: [],
   });
-
   // Fetching all the fetchedConnectorTypes and adding SVG its object to each connectorType and
   // then using unshift function to add an object for Imported Dataset to entire ConnectorTypes Array.
-  const getConnectorTypesNames = async () => {
-    let fetchedConnectorTypesFromAPI = await fetchConnectors();
-
-    fetchedConnectorTypesFromAPI = fetchedConnectorTypesFromAPI.map((connectorType) => {
-      if (connectorType.name === 'S3') {
-        return {
-          ...connectorType,
-          SVG: S3,
-        };
-      } else if (connectorType.name === 'Database') {
-        return {
-          ...connectorType,
-          SVG: Database,
-        };
-      } else if (connectorType.name === 'BigQuery') {
-        return {
-          ...connectorType,
-          SVG: BigQuery,
-        };
-      } else if (connectorType.name === 'GCS') {
-        return {
-          ...connectorType,
-          SVG: GCS,
-        };
-      } else if (connectorType.name === 'Spanner') {
-        return {
-          ...connectorType,
-          SVG: Spanner,
-        };
-      } else if (connectorType.name === 'Kafka') {
-        return {
-          ...connectorType,
-          SVG: Kafka,
-        };
-      } else if (connectorType.name === 'SQL Server') {
-        return {
-          ...connectorType,
-          SVG: SQLServer,
-        };
-      } else if (connectorType.name === 'MySQL') {
-        return {
-          ...connectorType,
-          SVG: MySQL,
-        };
-      } else if (connectorType.name === 'Oracle') {
-        return {
-          ...connectorType,
-          SVG: Oracle,
-        };
-      } else if (connectorType.name === 'PostgreSQL') {
-        return {
-          ...connectorType,
-          SVG: PostgreSQL,
-        };
-      } else if (connectorType.name === 'File') {
-        return {
-          ...connectorType,
-          SVG: ImportDatasetIcon,
-        };
-      } else if (connectorType.name === 'CloudSQLMySQL') {
-        return {
-          ...connectorType,
-          SVG: CloudSQLMySQL,
-        };
-      } else if (connectorType.name === 'CloudSQLPostgreSQL') {
-        return {
-          ...connectorType,
-          SVG: CloudSQLPostGreSQL,
-        };
-      } else {
-        return {
-          ...connectorType,
-          SVG: BigQuery,
-        };
-      }
-    });
-
-    fetchedConnectorTypesFromAPI.unshift({
-      name: 'Imported Datasets',
-      type: 'default',
-      category: 'default',
-      description: 'All Connections from the List',
-      artifact: {
-        name: 'allConnections',
-        version: 'local',
-        scope: 'local',
-      },
 
-      SVG: ImportDatasetIcon,
-    });
+  const classes = useStyles();
+  const connectorTypes: IConnectorArray[] = connectorsData.connectorTypes;
 
-    setConnectorTypes({
-      fetchedConnectorTypes: fetchedConnectorTypesFromAPI,
-    });
+  const updateState = (updatedState: { connectorTypes: IConnectorArray[] }) => {
```

Comment:
Why does this need to be passed as a callback? wouldn't it be less confusing if getWidgetData returned the widget data and you setConnectorsData there?

Reply:
modified the code by not passing updateState function as a callback.